lib/rules/jest/no-vague-titles.js

```diff
@@ -0,0 +1,92 @@
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Prevent the usage of vague words in test statements.',
+      category: 'Best Practices',
+      recommended: false,
+      uri:
+        'https://github.com/Shopify/eslint-plugin-shopify/blob/master/docs/rules/no-vague-titles.md',
+    },
+  },
+  create(context) {
+    const ignored = (context.options[0] && context.options[0].ignore) || [];
+
+    function isIgnoredFunctionName(node) {
+      if (node.type === 'MemberExpression') {
+        return ignored.some((method) => method === node.object.name);
+      }
+      return ignored.some((method) => method === node.callee.name);
+    }
+
+    function validate(node) {
+      if (
+        !isTestFunction(node) ||
+        isIgnoredFunctionName(node) ||
+        !containsDescription(node)
+      ) {
+        return;
+      }
+
+      const description = getDescription(node);
+
+      if (containsVagueWord(description)) {
+        context.report({
+          message: `{{ method }} statement should not contain vague words`,
+          data: {method: node.callee ? node.callee.name : node.object.name},
+          node,
+        });
+      }
+    }
+
+    return {
+      MemberExpression(node) {
+        validate(node);
+      },
+      CallExpression(node) {
+        validate(node);
+      },
+    };
+  },
+};
+
+function isTestFunction(node) {
+  if (node.type === 'MemberExpression') {
+    return node.object && matchTestFunctionName(node.object.name);
+  }
+
+  return node.callee && matchTestFunctionName(node.callee.name);
+}
+
+function matchTestFunctionName(functionName) {
+  return (
+    functionName === 'it' ||
+    functionName === 'fit' ||
+    functionName === 'test' ||
+    functionName === 'describe'
+  );
+}
+
+function containsDescription(node) {
+  const args = node.arguments || node.parent.arguments || [];
+  return (
+    args &&
+    args[0] &&
+    (args[0].type === 'Literal' || args[0].type === 'TemplateLiteral')
+  );
+}
+
+function getDescription(node) {
+  const args = node.arguments || node.parent.arguments || [];
+  const firstArgument = args[0];
+  const type = firstArgument.type;
+
+  if (type === 'Literal') {
+    return firstArgument.value;
+  }
+
+  return firstArgument.quasis[0].value.raw;
```

Comment:
This is a little odd, might want to merge together all the quasis that you can to come up with a complete description